src/language-js/embed.js

```diff
@@ -655,4 +657,15 @@ function printHtmlTemplateLiteral(path, print, textToDoc, parser, options) {
   );
 }
 
+function printTemplateExpression(print) {
+  return (path) => {
+    const node = path.getValue();
+    let printed = print(path);
+    if (node.comments && node.comments.length) {
+      printed = concat([indent(concat([softline, printed])), softline]);
+    }
+    return concat(["${", printed, lineSuffixBoundary, "}"]);
```

Comment:
I'm not totally sure that it's impossible for a descendant node's comment to get printed after the expression. There will be no indentation in that at case, but at least `lineSuffixBoundary` will prevent it from being printed after the closing brace.